classes/MultiStoreFixer.php

```diff
@@ -0,0 +1,165 @@
+<?php
+/**
+ * 2007-2020 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Academic Free License 3.0 (AFL-3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/AFL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2020 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License 3.0 (AFL-3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\Module\PrestashopCheckout;
+
+class MultiStoreFixer
+{
+    const UNIQUE_CONFIGURATION_KEYS = [
+        'PS_CHECKOUT_SHOP_UUID_V4',
+        'PS_CHECKOUT_PAYPAL_ID_MERCHANT',
+        'PS_PSX_FIREBASE_ID_TOKEN',
+        'PS_PSX_FIREBASE_LOCAL_ID',
+        'PS_PSX_FIREBASE_REFRESH_TOKEN',
+    ];
+
+    /**
+     * @return bool
+     */
+    public function run()
+    {
+        return \Shop::isFeatureActive() ? $this->fixMultiStore() : $this->fixSingleStore();
+    }
+
+    /**
+     * In single store mode, we remove global value and save only value for default shop
+     *
+     * @return bool
+     */
+    private function fixSingleStore()
+    {
+        $result = true;
+
+        foreach (self::UNIQUE_CONFIGURATION_KEYS as $key) {
+            // Get current value
+            $value = \Configuration::get($key);
+
+            // Remove current global value
+            \Configuration::deleteByName($key);
+
+            // Save current value associated to default shop
+            $result &= \Configuration::updateValue(
```

Comment:
The operator `&=` is actually a bitwise operator, which returns an integer.

When you deal with booleans, you should use the following syntax:

```suggestion
            $result = $result && \Configuration::updateValue(
```

This allows you to remove the `(bool)` at the end of the method.